src/python/pants/option/global_options.py

```diff
@@ -183,14 +187,24 @@ def from_options(cls, options: Options) -> ExecutionOptions:
             else:
                 remote_execution_headers = auth_plugin_result.execution_headers
                 remote_store_headers = auth_plugin_result.store_headers
+                if (
+                    remote_instance_name is not None
+                    and remote_instance_name != auth_plugin_result.instance_name
+                ):
+                    logger.debug(
+                        f"Overriding `--remote-instance-name={repr(remote_instance_name)}` to "
+                        f"instead be {repr(auth_plugin_result.instance_name)} due to the plugin "
+                        "from `--remote-auth-plugin`."
+                    )
+                remote_instance_name = auth_plugin_result.instance_name
```

Comment:
This line needs to be indented to fall within the `if` statement, correct?

Reply:
My thinking was to not log this if `--remote-instance-name` was unconfigured; this was originally only meant to warn that we're overriding something you set.

But this is a debug statement, so I'm not as concerned about it being noisy. It's probably good to know with debug logging what instance name was used either way.

Reply:
Right, but as it is now, `remote_instance_name` will always be overwritten even if `auth_plugin_result.instance_name` is `None`, correct?

Reply:
Yep, see the revised logic. We always log if the dynamically set value != the original one from the options system. 

Because this a debug level log, I think that's useful and not too noisy.